work_queue/src/work_queue.c

```diff
@@ -7416,4 +7415,259 @@ int work_queue_specify_min_taskid(struct work_queue *q, int minid) {
 	return q->next_taskid;
 }
 
+//the functions below are used by qsort in order to sort the work_queue_wsummary data
+static int comparecount(const void *a, const void *b)
+{
+	const struct work_queue_wsummary *x = b, *y = a;
+	return x->count - y->count;
+}
+static int comparecores(const void *a, const void *b)
+{
+	const struct work_queue_wsummary *x = b, *y = a;
+	return  ((x->cores - y->cores) == 0) ? 
+			(x->count - y->count) : (x->cores - y->cores);
+}
+static int comparememory(const void *a, const void *b)
+{
+	const struct work_queue_wsummary *x = b, *y = a;
+	return  ((x->memory - y->memory) == 0) ? 
+			(x->count - y->count) : (x->memory - y->memory);
+}
+static int comparedisk(const void *a, const void *b)
+{
+	const struct work_queue_wsummary *x = b, *y = a;
+	return  ((x->disk - y->disk) == 0) ? 
+			(x->count - y->count) : (x->disk - y->disk);
+}
+static int comparegpus(const void *a, const void *b)
+{
+	const struct work_queue_wsummary *x = b, *y = a;
+	return  ((x->gpus - y->gpus) == 0) ? 
+			(x->count - y->count) : (x->gpus - y->gpus);
+}
+
+// function used by other functions
+static void sort_work_queue_worker_summary(struct work_queue_wsummary data[], wsummary_sort_category sortby, int current_length)
+{
+	struct work_queue_wsummary *list_of_buckets = malloc(sizeof(struct work_queue_wsummary) * current_length); // create temporary data structure to sort list
+	for (int i = 0; i < current_length; i++)
+	{
+		list_of_buckets[i].count = data[i].count;	
+		list_of_buckets[i].cores = data[i].cores;	
+		list_of_buckets[i].memory = data[i].memory;	
+		list_of_buckets[i].gpus = data[i].gpus;		
+		list_of_buckets[i].disk = data[i].disk;	
+	}
+	switch(sortby)
+	{
+		case COUNT:
+			qsort(list_of_buckets, current_length, sizeof(struct work_queue_wsummary), comparecount);
+			break;
+		case CORES:
+			qsort(list_of_buckets, current_length, sizeof(struct work_queue_wsummary), comparecores);
+			break;
+		case MEMORY:
+			qsort(list_of_buckets, current_length, sizeof(struct work_queue_wsummary), comparememory);
+			break;
+		case DISK:
+			qsort(list_of_buckets, current_length, sizeof(struct work_queue_wsummary), comparedisk);
+			break;
+		case GPUS:
+			qsort(list_of_buckets, current_length, sizeof(struct work_queue_wsummary), comparegpus);
+			break;
+		case NONE:
+			break;
+	}
+	for (int i = 0; i < current_length; i++) // copy back sorted data
+	{
+		data[i].count = list_of_buckets[i].count;
+		data[i].cores = list_of_buckets[i].cores;
+		data[i].memory = list_of_buckets[i].memory;	
+		data[i].gpus = list_of_buckets[i].gpus;
+		data[i].disk = list_of_buckets[i].disk;	
+	}
+	free(list_of_buckets);
+}
+
+void convert_wsummary_to_log_scale(struct work_queue_wsummary worker_data[], int *current_length)
```

Comment:
Since this is a public part of the interface, call it `work_queue_wsummary_compact`